packages/field/test-suites/FormatMixin.suite.js

```diff
@@ -0,0 +1,394 @@
+    const options = { type: cfg.modelValueType, toggleValue: false, viewValue: false, ...opts };
+
+    switch (options.type) {
+      case Number:
+        return !options.toggleValue ? '123' : '456';
+      case Date:
+        // viewValue instead of modelValue, since a Date object is unparseable.
+        // Note: this viewValue needs to point to the same date as the
+        // default returned modelValue.
+        if (options.viewValue) {
+          return !options.toggleValue ? '5-5-2005' : '10-10-2010';
+        }
+        return !options.toggleValue ? new Date('5-5-2005') : new Date('10-10-2010');
+      case Array:
+        return !options.toggleValue ? ['foo', 'bar'] : ['baz', 'yay'];
+      case Object:
+        return !options.toggleValue ? { foo: 'bar' } : { bar: 'foo' };
+      case Boolean:
+        return !options.toggleValue;
+      case 'email':
+        return !options.toggleValue ? 'some-user@ing.com' : 'another-user@ing.com';
+      case 'iban':
+        return !options.toggleValue ? 'SE3550000000054910000003' : 'CH9300762011623852957';
+      default:
+        return !options.toggleValue ? 'foo' : 'bar';
+    }
+  }
+
+  describe(`FormatMixin ${cfg.suffix ? `(${cfg.suffix})` : ''}`, async () => {
+    let elem;
+    let nonFormat;
+    let fooFormat;
+
+    before(async () => {
+      if (!cfg.tagString) {
+        cfg.tagString = defineCE(
+          class extends FormatMixin(LitElement) {
+            render() {
+              return html`
+                <slot name="input"></slot>
+              `;
+            }
+
+            set value(newValue) {
+              this.inputElement.value = newValue;
+            }
+
+            get value() {
+              return this.inputElement.value;
+            }
+
+            get inputElement() {
+              return this.querySelector('input');
+            }
+          },
+        );
+      }
+
+      elem = unsafeStatic(cfg.tagString);
+      nonFormat = await fixture(html`<${elem}
+        .formatter="${v => v}"
+        .parser="${v => v}"
+        .serializer="${v => v}"
+        .deserializer="${v => v}"
+      ><input slot="input">
+      </${elem}>`);
+      fooFormat = await fixture(html`
+      <${elem}
+        .formatter="${value => `foo: ${value}`}"
+        .parser="${value => value.replace('foo: ', '')}"
+        .serializer="${value => `[foo] ${value}`}"
+        .deserializer="${value => value.replace('[foo] ', '')}"
+      ><input slot="input">
+      </${elem}>`);
+    });
+
+    it('fires `model-value-changed` for every change on the input', async () => {
+      const formatEl = await fixture(html`<${elem}><input slot="input"></${elem}>`);
+
+      let counter = 0;
+      formatEl.addEventListener('model-value-changed', () => {
+        counter += 1;
+      });
+
+      mimicUserInput(formatEl, generateValueBasedOnType());
+      expect(counter).to.equal(1);
+
+      // Counter offset +1 for Date because parseDate created a new Date object
```

Comment:
Maybe add a TODO here that the ideal fix would be to fix this in the hasChanged of the date inputs?